AE-1329 Add link to superseding energiatodistus

muep opened the pull request.

muep committed:
AE-1329 Add link to superseding energiatodistus

muep commented:
Tästä puuttuu vielä se korvaus(kö?)ajankohdan näyttäiminen.

muep commented:
Semmoinen ajatus tässä tuli, että tuo `korvaava-energiatodistus-id` taitaa löytyä jo siinä vaiheessa, kun on luonnos tekeillä. Ehkä pitäisi vähän tarkempi käsittely tehdä tuonne, mutta en vielä testannut asiaa.

muep commented:
Testailun perusteella selvisi, että kyllä tuo korvaavuus näkyy `korvaava-energiatodistus-id`:ssa jo siinä vaiheessa, kun se on vasta luonnos.

muep committed:
AE-1329 Display allekirjoitusaika for korvaava ET

muep commented:
Nyt tämä on opetettu hakemaan se korvaava energiatodistus, ja siitä voidaan tarkistaa allekirjoitettuus ja allekirjoituksen ajankohta.

muep commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -39,6 +41,28 @@
 
   $: load(params);
 
+  // whoami -> versio -> id -> Future {energiatodistus: ET, korvattuEnergiatodistus: Maybe ET}
```
tämä kommentti oli lähinnä itselle ajatusten kasaamiseksi, voisi varmaan siivota pois

muep commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -39,6 +41,28 @@
   $: load(params);
 
+  // whoami -> versio -> id -> Future {energiatodistus: ET, korvattuEnergiatodistus: Maybe ET}
+  const fetchEnergiatodistus = (whoami, versio, id) => {
```
tuntui helpommalta tehdä jonkin verran imperatiivisen kautta, mutta tarvittaessa voidaan kyllä jäsennellä funktionaalisemmaksikin

hkorpi replied:
joo kyllä tätä kannattaa vielä hioa, koska esim. maybe tyyppi näin käytettynä ei oikeastaan eroa nullista yhtään

hkorpi replied:
kokeile vielä vähän - tein itse tästä myös oman version

muep commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -162,6 +183,23 @@
+        <div>
+          {#each korvaavaEnergiatodistus
+            .toArray()
+            .filter(R.propEq('tila-id', 2)) as korvaavaEt}
```
Vaihtoehtoisesti voisi katsoa että allekirjoitusaika löytyy. Silloin tulisi myös muita kuin allekirjoitettu tila sellaisiksi, että linkki näytetään.

hkorpi replied:
tilojen tai luokkien id arvoja ei saa käyttää suoraan vaan pitää etsiä sopiva predikaattifunktio

hkorpi replied:
et utilssa on isSigned

muep replied:
isSigned on nyt käytössä

hkorpi commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -39,6 +41,28 @@
+
+      return R.map(
+        Maybe.Some,
+        EnergiatodistusApi.getEnergiatodistusById('all', korvaavaEtId.some())
```
some metodia ei ole tarkoitettu käytettäväksi

hkorpi commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -39,6 +41,28 @@
+
+    return { energiatodistus, korvaavaEnergiatodistus };
+  };
+
```
tässä on luultavasti vielä se ongelma että tämä hakee energiatodistuksen kahteen kertaan
katsoppas network välilehdeltä?

muep committed:
AE-1329 Use cached future for the initial ET

muep committed:
AE-1329 More functional way to express korvaavaEnergiatodistus

muep committed:
AE-1329 Use fold instead of the nested maps and orSome

muep committed:
AE-1329 Avoid matching tila-id against magic number

hkorpi commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -162,6 +179,23 @@
+            .toArray()
+            .filter(ETUtils.isSigned) as korvaavaEt}
+            <span
+              >{Maybe.fold(
```
välilyönti siistii automaattista formatointia, jos ei seuraa mitään ongelmaa leiskaan

muep replied:
Toisaan paljon nätimpi tuli pisti whitespacea mukaan.

hkorpi commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -39,6 +42,23 @@
+        R.prop('korvaava-energiatodistus-id') // Maybe<id>
+      ),
+      energiatodistus
+    );
```
:+1:

hkorpi commented on src/pages/valvonta-oikeellisuus/valvonta.svelte:
```diff
@@ -162,6 +179,23 @@
+              {i18n('energiatodistus.korvaavuus.header.korvaava')}</span>
+            <Link
+              bold={true}
+              href={'/#/energiatodistus/' + korvaavaEt.id}
```
laita tähän myös versio (eli suora linkki versiokohtaiselle sivulle) niin ei mene turhaan resolverin kautta - resolveri aina lataa tiedot

hkorpi replied:
resolverin kautta on tarkoitus mennä vain jos versiota ei ole tiedossa

muep replied:
Nyt pitäisi olla versiollinen

muep committed:
AE-1329 Adjust code formatting

Prettier did not seem to dare to add whitespace between the tag and
its contents.

muep committed:
AE-1329 ET link with version number included
